src/main/java/uk/gov/justice/hmpps/prison/service/digitalwarrant/DigitalWarrantService.java

```diff
@@ -0,0 +1,144 @@
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenceResult;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderBooking;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderCharge;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderCourtCase;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderSentence;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderSentenceCharge;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.SentenceCalcType;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.SentenceTerm;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.AgencyLocationRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenceRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenceResultRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderBookingRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderChargeRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderCourtCaseRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderSentenceChargeRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderSentenceRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.ReferenceCodeRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.SentenceCalcTypeRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.SentenceTermRepository;
+import uk.gov.justice.hmpps.prison.service.EntityNotFoundException;
+
+import javax.transaction.Transactional;
+
+import java.util.Comparator;
+
+import static java.util.Arrays.asList;
+
+@Service
+public class DigitalWarrantService {
+    @Autowired
+    private OffenderCourtCaseRepository offenderCourtCaseRepository;
+    @Autowired
+    private OffenderChargeRepository offenderChargeRepository;
+    @Autowired
+    private OffenderSentenceRepository offenderSentenceRepository;
+    @Autowired
+    private AgencyLocationRepository agencyLocationRepository;
+    @Autowired
+    private OffenceRepository offenceRepository;
+    @Autowired
+    private OffenderBookingRepository offenderBookingRepository;
+    @Autowired
+    private SentenceCalcTypeRepository sentenceCalcTypeRepository;
+    @Autowired
+    private OffenderSentenceChargeRepository offenderSentenceChargeRepository;
+    @Autowired
+    private ReferenceCodeRepository<LegalCaseType> legalCaseTypeReferenceCodeRepository;
+    @Autowired
+    private OffenceResultRepository offenceResultRepository;
+
+    @Autowired
+    private SentenceTermRepository sentenceTermRepository;
+
+    @Transactional
+    public Long createCourtCase(Long bookingId, CourtCase courtCase) {
+        AgencyLocation agency = agencyLocationRepository.findById(courtCase.getAgencyId()).orElseThrow(EntityNotFoundException.withId(courtCase.getAgencyId()));
```

Comment:
Vey minor style observation: I think we would prefer uses of `var`

Reply:
Thanks I'll make a note of this on the improvement ticket.